sdk/communication/Azure.Communication.Administration/src/Models/ReleasePhoneNumberOperation.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication.Administration.Models
+{
+    /// <summary>
+    /// Represents a long-running release phone number operation.
+    /// </summary>
+    public class ReleasePhoneNumberOperation : Operation<PhoneNumberRelease>
+    {
+        private readonly object _lockObject = new object();
+        private readonly PhoneNumberAdministrationClient _client;
+        private readonly CancellationToken _cancellationToken;
+        private bool _hasCompleted;
+        private PhoneNumberRelease? _value;
+        private Response _rawResponse;
+        private Response _finalRawResponse;
+
+        /// <summary>
+        /// Initializes a new <see cref="ReleasePhoneNumberOperation"/> instance
+        /// </summary>
+        /// <param name="client">The client used to check for completion.</param>
+        /// <param name="id">The phone number release operation ID.</param>
+        /// <param name="cancellationToken">The cancellation token to use.</param>
+        public ReleasePhoneNumberOperation(
+            PhoneNumberAdministrationClient client,
+            string id,
+            CancellationToken cancellationToken = default)
+        {
+            Id = id;
+            _value = null;
+            _rawResponse = null!;
+            _finalRawResponse = null!;
+            _client = client;
+            _cancellationToken = cancellationToken;
+        }
+
+        /// <summary>
+        /// Initializes a new <see cref="ReleasePhoneNumberOperation"/> instance
+        /// </summary>
+        /// <param name="client">The client used to check for completion.</param>
+        /// <param name="id">The phone number release operation ID.</param>
+        /// <param name="initialResponse">The original server response on start operation request.</param>
+        /// <param name="cancellationToken">The cancellation token to use.</param>
+        internal ReleasePhoneNumberOperation(
+            PhoneNumberAdministrationClient client,
+            string id,
+            Response initialResponse,
+            CancellationToken cancellationToken = default)
+        {
+            Id = id;
+            _value = null;
+            _rawResponse = initialResponse;
+            _finalRawResponse = null!;
+            _client = client;
+            _cancellationToken = cancellationToken;
+        }
+
+        /// <inheritdocs />
+        public override string Id { get; }
+
+        /// <inheritdocs />
+        public override PhoneNumberRelease Value => OperationHelpers.GetValue(ref _value);
+
+        /// <inheritdocs />
+        public override bool HasCompleted => _hasCompleted;
```

Comment:
Can be an auto-property with private set

Reply:
It's override for get, but there is not set, so it does not compile then